pageserver/src/tenant/mgr.rs

```diff
@@ -262,14 +268,31 @@ pub async fn shutdown_all_tenants() {
     for (tenant_id, tenant) in tenants_to_shut_down {
         join_set.spawn(
             async move {
+                let (_guard, shutdown_progress) = completion::channel();
                 let freeze_and_flush = true;
+                if let Err(progress) = tenant.shutdown(shutdown_progress, freeze_and_flush).await {
```

Comment:
You added a `warn` for slow shutdown in the coalesced case. Let's have a `warn` for the first shutdown call as well, if it's taking longer than 5 seconds. Hm, we'd need that `warn` inside the `Tenant::shutdown` impl since the function is invoked in multiple places.

Reply:
So what will the warn warn about? Uploading is taking time?

Reply:
Doesn't have to be specific. Just to leave some breadcrumbs in the logs to understand why a tenant/pageserver is failing to shut down in time.

Reply:
You commented elsewhere so I assuming this is fine.

Reply:
This created a huge storm and disabled our logging in prod as staging.

Fix was in #4886. Basically the rule is "do not log anything per timeline or per tenant" in shutdown.